ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/admin/party_management_service.proto

```diff
@@ -0,0 +1,133 @@
+  // This request will always succeed.
+  // TODO: Check what the use case is for listing non-local parties
+  rpc ListKnownParties (ListKnownPartiesRequest) returns
+    (ListKnownPartiesResponse);
+
+  // Adds a new party to the set managed by the backing participant.
+  // Caller specifies a party identifier suggestion, the actual identifier
+  // allocated might be different and is implementation specific.
+  // This call will either succeed or respond with ALREADY_EXISTS if given party
+  // already exists and the party allocation scheme uses only request data to 
+  // generate new stable identifiers.
+  // daml-on-sql: suggestion's uniqueness is checked and call rejected if the
+  // identifier is already present
+  // daml-on-kv-ledger: suggestion's uniqueness is checked bby the validators in
+  // the consensus layer and call rejected if the identifier is already present.
+  // canton: completely different globally unique identifier is allocated.
+  // Behind the scenes calls to an internal protocol are made. As that protocol
+  // is richer than the the surface protocol, the arguments take implicit values
+  rpc AllocateParty (AllocatePartyRequest) returns (AllocatePartyResponse);
+}
+
+message GetParticipantIdRequest {
+
+  // Must correspond to the ledger ID reported by the Ledger Identification
+  // Service.
+  // Required
+  string ledger_id = 1;
+}
+
+message GetParticipantIdResponse {
+
+  // Identifier of the participant, which SHOULD be globally unique.
+  string participant_id = 2;
+}
+
+message ListKnownPartiesRequest {
+
+  // Must correspond to the ledger ID reported by the Ledger Identification
+  // Service.
+  // Required
+  string ledger_id = 1;
+}
+
+message ListKnownPartiesResponse {
+
+  // The details of all DAML parties hosted by the participant.
+  // Required
+  repeated PartyDetails party_details = 1;
+}
+
+message PartyDetails {
+
+  // The stable unique identifier of a DAML party.
+  // Required
+  string party = 1;
+
+  // Common name associated with the party.
+  // Optional
+  string common_name = 2;
+
+  // id of the participant that this party is hosted on
+  // Required
+  string participant_id = 3;
+
+  // Future extension: include metadata fields. There will be two categories
+  // of such fields: common and implementation specific.
+  // CommonMetadata common_metadata = 2;
+  // bytes implementation_metadata = 3;
+  
+  // Canton-specific meta-data would contain the following:
+  // string domain = 2;
+  // string participant = 3;
+  // com.digitalasset.canton.identity.v0.ParticipantPrivilege privilege = 4;
+  // com.digitalasset.canton.identity.v0.TrustLevel trustLevel = 5;
+}
+
+message AllocatePartyRequest {
+
+  // Must correspond to the ledger ID reported by the Ledger Identification
+  // Service.
+  // Required
+  string ledger_id = 1;
+
+  // Common name of the party to be added to the participant.
+  // Required
+  string common_name = 2;
```

Comment:
Having read https://support.dnsimple.com/articles/what-is-common-name/ I believe "common_name" gives too much of an impression of uniqueness or validatedness. I'd rather be on the safe side and use

```
// Human readable name. Might not be unique.
string display_name
```